pandas/core/reshape/merge.py

```diff
@@ -1730,6 +1730,12 @@ def flip(xs) -> np.ndarray:
         )
         tolerance = self.tolerance
 
+        if is_object_dtype(right_values):
+            raise ValueError("Right input is not from numerical dtype")
+
+        if is_object_dtype(left_values.dtype):
+            raise ValueError("Left input is not from numerical dtype")
```

Comment:
I think this should be a TypeError and not a ValueError. Also how about combining these into a single check / error (something about both needing to be numeric)? Otherwise there's an asymmetry in the error that gets raised (e.g., if both are object then you'd only get an error message about the right but not the left input).

Reply:
Yeah that is a good point. Changed it accordingly. Used a ValueError because that was proposed in the issue. Changed it to TypeError.